src/node_api.h

```diff
@@ -73,16 +78,39 @@ typedef struct napi_module {
   static void fn(void)
 #endif
 
+#ifdef NAPI_EXPERIMENTAL
+#ifdef NODE_API_CUSTOM_FEATURES
+
+// Define value of node_api_module_features variable in your module when
+// NODE_API_CUSTOM_FEATURES is set in gyp file.
+extern node_api_features node_api_module_features;
+#define NODE_API_DEFINE_DEFAULT_FEATURES
+
+#else  // NODE_API_CUSTOM_FEATURES
+
+#define NODE_API_DEFINE_DEFAULT_FEATURES                                       \
+  static node_api_features node_api_module_features = node_api_default_features;
+
+#endif  // NODE_API_CUSTOM_FEATURES
+
+#define NODE_API_FEATURES_PTR /* NOLINT */ &node_api_module_features,
+
+#else  // NAPI_EXPERIMENTAL
+#define NODE_API_DEFINE_DEFAULT_FEATURES
+#define NODE_API_FEATURES_PTR
+#endif  // NAPI_EXPERIMENTAL
+
 #define NAPI_MODULE_X(modname, regfunc, priv, flags)                           \
```

Comment:
Should we just create a new NAPI_MODULE_X_FEATURES which takes the features parameter?

Reply:
It may be simpler - let me try.